src/dstore/pmix_esh.c

```diff
@@ -130,14 +138,64 @@ static inline const char *_unique_id(void)
     return str;
 }
 
-int _esh_init(void)
+int _esh_init(pmix_info_t info[], size_t ninfo)
 {
     int rc;
     seg_desc_t *seg;
+    size_t n;
 
     PMIX_OUTPUT_VERBOSE((10, pmix_globals.debug_output,
                          "%s:%d:%s", __FILE__, __LINE__, __func__));
 
+    /* find the temp dir */
+    if (_is_server()) {
+        if (NULL == (_base_path = getenv(PMIX_SERVER_TMPDIR_ENV))) {
+            if (NULL == (_base_path = getenv("TMPDIR"))) {
+                if (NULL == (_base_path = getenv("TEMP"))) {
+                    if (NULL == (_base_path = getenv("TMP"))) {
+                        _base_path = "/tmp";
+                    }
+                }
+            }
+        }
+
+        /* scan incoming info for directives */
+        if (NULL != info) {
+            for (n=0; n < ninfo; n++) {
+                if (0 == strcmp(PMIX_USERID, info[n].key)) {
+                    jobuid = info[n].value.data.uint32;
+                    setjobuid = 1;
+                    continue;
+                }
+                if (0 == strcmp(PMIX_DSTPATH, info[n].key)) {
+                    if(_nspace_path != NULL)
```

Comment:
we set _base_path here as well.
This setting has the priority, we shouldn't ignore PMIX_SERVER_TMPDIR if we already saw PMIX_DSTPATH.